benches/trap.rs

```diff
@@ -0,0 +1,146 @@
+                                    for _ in 0..ITERS {
+                                        let mut store = Store::new(&engine, ());
+                                        let instance =
+                                            Instance::new(&mut store, &module, &[]).unwrap();
+                                        let f = instance
+                                            .get_typed_func::<(), (), _>(&mut store, "")
+                                            .unwrap();
+                                        assert!(f.call(&mut store, ()).is_err());
+                                    }
+                                }
+                            })
+                        })
+                        .collect::<Vec<_>>();
+                    threads.into_iter().for_each(|t| t.join().unwrap());
+                });
+            },
+        );
+    }
+
+    group.finish();
+}
+
+fn bench_many_modules_registered_traps(c: &mut Criterion) {
+    let mut group = c.benchmark_group("many-modules-registered-traps");
+
+    for num_modules in vec![1, 8, 64, 512, 4096] {
+        group.throughput(Throughput::Elements(num_modules));
+        group.bench_with_input(
+            BenchmarkId::from_parameter(num_modules),
+            &num_modules,
+            |b, &num_modules| {
+                let engine = Engine::default();
+                let modules = (0..num_modules)
+                    .map(|_| module(&engine, 10).unwrap())
+                    .collect::<Vec<_>>();
+
+                b.iter_custom(|iters| {
+                    let mut total = std::time::Duration::default();
+
+                    for _ in 0..iters {
+                        let mut store = Store::new(&engine, ());
+                        let instance =
+                            Instance::new(&mut store, modules.last().unwrap(), &[]).unwrap();
+                        let f = instance
+                            .get_typed_func::<(), (), _>(&mut store, "")
+                            .unwrap();
+
+                        let start = std::time::Instant::now();
+                        assert!(f.call(&mut store, ()).is_err());
+                        total += start.elapsed();
+                    }
+
+                    total
+                });
+            },
+        );
+    }
+
+    group.finish()
+}
+
+fn bench_many_stack_frames_traps(c: &mut Criterion) {
+    let mut group = c.benchmark_group("many-stack-frames-traps");
+
+    for num_stack_frames in vec![1, 8, 64, 512] {
+        group.throughput(Throughput::Elements(num_stack_frames));
+        group.bench_with_input(
+            BenchmarkId::from_parameter(num_stack_frames),
+            &num_stack_frames,
+            |b, &num_stack_frames| {
+                let engine = Engine::default();
+                let modules = (0..num_stack_frames)
+                    .map(|_| module(&engine, num_stack_frames).unwrap())
+                    .collect::<Vec<_>>();
```

Comment:
I think this may be a copy/paste error and there should only be one module in this benchmark?

Reply:
Good eye.

Reply:
Fixed.